index.js

```diff
@@ -118,93 +149,146 @@ async function isNotLatestTerraformVersion() {
+    return `${projectName}\n\n${strCommits}`;
+  });
+  logs = await Promise.all(logs);
+
+  logs = logs.join("\n\n");
+  if (await isNotLatestManifestsVersion()) {
+    logs = `⚠️ **The production version of manifests is behind the latest staging version. Consider upgrading to the latest version before merging this pull request.** \n\n ${logs}`;
+  }
+
+  if (await isNotLatestTerraformVersion()) {
+    logs = `⚠️ **The production version of the Terraform infrastructure is behind the latest staging version. Consider upgrading to the latest version before merging this pull request.** \n\n ${logs}`;
+  }
+  return logs;
+}
+
+async function getContents(owner, repo, path) {
   const { data: data } = await octokit.repos.getContents({
+    owner: owner,
+    repo: repo,
+    path: path,
+  });
+  return data;
+}
+
+async function createPR(
+  projects,
+  issueContent,
+  releaseConfig,
+  newReleaseContentBlob
+) {
+  const branchName = `release-${new Date().getTime()}`;
+  const manifestsSha = await getHeadSha("notification-manifests", "main");
+  const logs = await buildLogs(projects);
+
+  const ref = await octokit.git.createRef({
     owner: "cds-snc",
     repo: "notification-manifests",
-    path: "env/production/kustomization.yaml",
+    ref: `refs/heads/${branchName}`,
+    sha: manifestsSha,
   });
 
-  const { data: issueData } = await octokit.repos.getContents({
+  const manifestUpdates = projects
+    .map((project) => {
+      return `${project.name}:${shortSha(project.headSha)}`;
+    })
+    .join(" and ");
+  const update = await octokit.repos.createOrUpdateFile({
     owner: "cds-snc",
     repo: "notification-manifests",
-    path: ".github/PULL_REQUEST_TEMPLATE.md",
+    branch: branchName,
+    sha: releaseConfig.sha,
+    path: "env/production/kustomization.yaml",
+    message: `Updated manifests to ${manifestUpdates}`,
+    content: newReleaseContentBlob,
   });
 
-  const apiSha = await getHeadSha("notification-api");
-  const adminSha = await getHeadSha("notification-admin");
-  const manifestsSha = await getHeadSha("notification-manifests", "main");
-
-  const fileContent = YAML.parse(Base64.decode(data.content));
-  const issueContent = Base64.decode(issueData.content);
-
-  fileContent.images.forEach((image) => {
-    if (image.name == "admin") {
-      oldAdminSha = image.newName.split(":").slice(-1)[0]; // eslint-disable-line no-unused-vars, no-undef
-      image.newName = `${AWS_ECR_URL}/notify-admin:${adminSha.slice(0, 7)}`;
-    }
-    if (image.name == "api") {
-      oldApiSha = image.newName.split(":").slice(-1)[0]; // eslint-disable-line no-unused-vars, no-undef
-      image.newName = `${AWS_ECR_URL}/notify-api:${apiSha.slice(0, 7)}`;
-    }
+  const pr = await octokit.pulls.create({
+    owner: "cds-snc",
+    repo: "notification-manifests",
+    title: `[AUTO-PR] Automatically generated new release ${new Date().toISOString()}`,
+    head: branchName,
+    base: "main",
```

Comment:
Would be nice to reuse data from `PROJECTS`

Reply:
We could expand `PROJECTS` later on in a future iteration. We'd need to have another variable or a way to distinguish candidate projects to release versus "meta release projects" such as `notification-manifests`.